bofire/data_models/domain/constraints.py

```diff
@@ -9,23 +19,27 @@
 from bofire.data_models.constraints.api import AnyConstraint, Constraint
 from bofire.data_models.filters import filter_by_class
 
+C = TypeVar("C", bound=Union[AnyConstraint, Constraint])
```

Comment:
Trailing underscores are recommended to be used to avoid conflicts Python builtins, see https://peps.python.org/pep-0008/#descriptive-naming-styles. I would use something else, e.g., numbers to to have multiple constraint types.

Reply:
Yes that's fair! Although just to note, TypeVar has some [additional naming rules](https://stackoverflow.com/q/74589610), which I don't think allow numbers.